include/gtirb/Allocator.hpp

```diff
@@ -30,6 +30,7 @@
 #include <utility>
 #include <vector>
 
+namespace gtirb {
```

Comment:
It looks like there are tests that do not build with `-DGTIRB_WRAP_UTILS_IN_NAMESPACE=ON`:

```
[ 76%] Building CXX object src/test/CMakeFiles/TestGTIRB.dir/Allocator.test.cpp.o
.../gtirb/src/test/Allocator.test.cpp:33:19: error: ‘SpecificBumpPtrAllocator’ does not name a type
   33 | using Allocator = SpecificBumpPtrAllocator<AllocTest>;
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~
.../gtirb/src/test/Allocator.test.cpp:35:35: error: ‘Allocator’ has not been declared
   35 | inline void* operator new(size_t, Allocator& A) { return A.Allocate(); }
      |                                   ^~~~~~~~~
.../gtirb/src/test/Allocator.test.cpp: In function ‘void* operator new(size_t, int&)’:
.../gtirb/src/test/Allocator.test.cpp:35:60: error: request for member ‘Allocate’ in ‘A’, which is of non-class type ‘int’
   35 | inline void* operator new(size_t, Allocator& A) { return A.Allocate(); }
      |                                                            ^~~~~~~~
.../gtirb/src/test/Allocator.test.cpp: At global scope:
.../gtirb/src/test/Allocator.test.cpp:36:36: error: ‘Allocator’ has not been declared
   36 | inline void operator delete(void*, Allocator&) {}
      |                                    ^~~~~~~~~
.../gtirb/src/test/Allocator.test.cpp: In member function ‘virtual void Unit_Allocator_allocate_Test::TestBody()’:
.../gtirb/src/test/Allocator.test.cpp:40:3: error: ‘Allocator’ was not declared in this scope
   40 |   Allocator A;
      |   ^~~~~~~~~
In file included from .../gtirb/build/googletest-src/googletest/include/gtest/gtest.h:375,
                 from .../gtirb/src/test/Allocator.test.cpp:18:
.../gtirb/src/test/Allocator.test.cpp:41:18: error: ‘A’ was not declared in this scope
   41 |   EXPECT_NE(new (A) AllocTest, nullptr);
      |                  ^
.../gtirb/src/test/Allocator.test.cpp:42:18: error: ‘A’ was not declared in this scope
   42 |   EXPECT_NE(new (A) AllocTest, nullptr);
      |                  ^
.../gtirb/src/test/Allocator.test.cpp:43:18: error: ‘A’ was not declared in this scope
   43 |   EXPECT_NE(new (A) AllocTest, nullptr);
      |                  ^
.../gtirb/src/test/Allocator.test.cpp: In member function ‘virtual void Unit_Allocator_deallocate_Test::TestBody()’:
.../gtirb/src/test/Allocator.test.cpp:52:7: error: ‘Allocator’ was not declared in this scope
   52 |       Allocator A;
      |       ^~~~~~~~~
In file included from .../gtirb/build/googletest-src/googletest/include/gtest/gtest.h:375,
                 from .../gtirb/src/test/Allocator.test.cpp:18:
.../gtirb/src/test/Allocator.test.cpp:53:22: error: ‘A’ was not declared in this scope
   53 |       EXPECT_NE(new (A) AllocTest, nullptr);
      |                      ^
.../gtirb/src/test/Allocator.test.cpp:54:22: error: ‘A’ was not declared in this scope
   54 |       EXPECT_NE(new (A) AllocTest, nullptr);
      |                      ^
.../gtirb/src/test/Allocator.test.cpp:55:22: error: ‘A’ was not declared in this scope
   55 |       EXPECT_NE(new (A) AllocTest, nullptr);
      |                      ^
make[2]: *** [src/test/CMakeFiles/TestGTIRB.dir/build.make:90: src/test/CMakeFiles/TestGTIRB.dir/Allocator.test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:528: src/test/CMakeFiles/TestGTIRB.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

I think this should be a matter of adding the `gtirb::` namespace prefix in the `Allocator.test.cpp` file (and there may be more failures after that is resolved, I didn't try it).

Reply:
I set ` -DGTIRB_ENABLE_TESTS=OFF ` when building gtirb, thus not noticing that. Adding `ifdef`s to the tests can may solve the problems.

Reply:
Do the tests really need `#ifdef`? I was thinking updating uses of things to use the namespace. For example at `Allocator.test.cpp:33`:

```
using Allocator = SpecificBumpPtrAllocator<AllocTest>;
```

to

```
using Allocator = gtirb::SpecificBumpPtrAllocator<AllocTest>;
```

which will work regardless of the `GTIRB_WRAP_UTILS_IN_NAMESPACE` setting. 

It looks like this is the only change that needs to be made; everything builds fine with this change.

Reply:
This is certainly feasible as well.